lib/Telemetry/Monitor.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * @author Thomas Müller <thomas.mueller@tmit.eu>
+ *
+ * @copyright Copyright (c) 2018, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\ConfigReport\Telemetry;
+
+use OC\License\LicenseFetcher;
+use OCA\ConfigReport\ReportDataCollector;
+use OCP\Http\Client\IClientService;
+use OCP\IConfig;
+use OCP\ILogger;
+
+class Monitor {
+	private IConfig $config;
+	private ILogger $logger;
+	private IClientService $clientService;
+	private ReportDataCollector $collector;
+
+	public function __construct(IConfig $config, ILogger $logger, IClientService $clientService, ReportDataCollector $collector) {
+		$this->config = $config;
+		$this->logger = $logger;
+		$this->clientService = $clientService;
+		$this->collector = $collector;
+	}
+
+	public function sendTelemetry(): void {
+		$telemetryOptIn = $this->config->getSystemValue('telemetry.enabled', true);
+		if (!$telemetryOptIn) {
+			return;
+		}
+
+		$data = $this->collectTelemetryData();
+
+		$client = $this->clientService->newClient();
+		try {
+			$client->post('https://3qyh826rmd.execute-api.ap-southeast-1.amazonaws.com/default/pocTelemetry', [
```

Comment:
Should we make it configurable?
The use case would be, if we ever need to change the location, old versions would still send the data to the old endpoint without a chance to send it to the right one.

Reply:
Working for a US based storage company in the past having a similar telemetry report (call home) function, you must have an unchangeable target address. But I propose to have in addition a configurable value where you can add multiple other addresses like via email. So checking if email is configured, you send additional reports to addresses defined in that key. Topic: "Telemetry report from <ID>". Just my 2c